apps/modernization-ui/src/apps/page-builder/components/CreateQuestion/CreateQuestion.tsx

```diff
@@ -142,61 +141,89 @@ export const CreateQuestion = ({ modalRef, question }: any) => {
+                // requiredInMessage: data.requiredInMessage,
+                // hl7DataType: data.hl7DataType
             },
-            label: questionData.label,
-            uniqueName: questionData.uniqueName,
-            uniqueId: questionData.uniqueId,
-            subgroup: questionData.subgroup,
-            adminComments: questionData.adminComments,
-            description: questionData.label,
-            type: questionData.type,
-            codeSet: questionData.codeSet,
-            tooltip: questionData.tooltip || '',
-            displayControl: questionData.displayControl || 0,
-            defaultValue: questionData.defaultValue,
-            allowFutureDates: questionData.allowFutureDates,
-            fieldLength: questionData.fieldLength,
-            fieldSize: questionData.fieldSize,
-            mask: questionData.mask
-        };
-        // TODO based on the type selected call one of the questionAPI methods
-        // such as createDateQuestion()
-        console.log('ensure', request);
-    };
-    const handleUpdateQuestion = () => {
-        const request = {
-            ...questionData,
-            tooltip: questionData.tooltip || ''
+            adminComments: data.adminComments!,
+            codeSet: data.codeSet,
+            description: data.description,
+            defaultValue: data.defaultValue, // text
+            displayControl: parseInt(String(data.displayControl!), 10),
+            label: data.label,
+            mask: data.mask,
+            subgroup: data.subgroup,
+            tooltip: data.tooltip,
+            uniqueName: data.uniqueName,
+            uniqueId: data.uniqueId
         };
+        if (selectedFieldType === 'TEXT') {
+            QuestionControllerService.createTextQuestionUsingPost({
+                authorization,
+                request: request
+            }).then((response) => {
+                showAlert({ type: 'success', header: 'Created', message: 'Question created successfully' });
+                resetInput();
+                return response;
+            });
+        } else if (selectedFieldType === 'DATE') {
+            const dateRequest = { ...request, allowFutureDates: data.allowFutureDates === 'Yes' };
+            QuestionControllerService.createDateQuestionUsingPost({
+                authorization,
+                request: dateRequest
+            }).then((response) => {
+                showAlert({ type: 'success', header: 'Created', message: 'Question created successfully' });
+                resetInput();
+                return response;
+            });
+        } else if (selectedFieldType === 'NUMERIC') {
+            const numericRequest = {
+                ...request,
+                minValue: 0,
+                fieldLength: parseInt(String(data.fieldLength!), 10),
+                maxValue: 50,
+                relatedUnitsLiteral: 'ML',
+                relatedUnitsValueSet: 2,
+                defaultValue: 0
```

Comment:
Question: Isn't this hard-coding the values for the Numeric question instead of accepting the user input?